pkg/distributor/distributor.go

```diff
@@ -598,9 +599,12 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 	validatedSamples := 0
 	validatedExemplars := 0
 
-	if d.limits.AcceptHASamples(userID) && len(req.Timeseries) > 0 {
-		cluster, replica := findHALabels(d.limits.HAReplicaLabel(userID), d.limits.HAClusterLabel(userID), req.Timeseries[0].Labels)
-		removeReplica, err = d.checkSample(ctx, userID, cluster, replica)
+	// Cache user limit with overrides.
```

Comment:
Just a nit pick. I think it would be a more useful comment if it mention why caching is important. For example `// Cache user limit with overrides so we spend less CPU doing locking. See issue #4904`